src/v2/coins/eth.js

```diff
@@ -329,7 +383,62 @@ class Eth extends BaseCoin {
     }
   }
 
+  getAddressNonce(address, callback) {
+    return co(function *() {
+      // Get nonce for backup key (should be 0)
+      let nonce = 0;
+
+      const result = yield this.recoveryBlockchainExplorerQuery({
+        module: 'account',
+        action: 'txlist',
+        address
+      });
+      const backupKeyTxList = result.result;
+      if (backupKeyTxList.length > 0) {
+        // Calculate last nonce used
+        const outgoingTxs = backupKeyTxList.filter((tx) => tx.from === address);
+        nonce = outgoingTxs.length;
+      }
+
+      return nonce;
+    }).call(this).asCallback(callback);
+  }
+
+
   /**
+   * Helper function for recover()
+   * This transforms the unsigned transaction information into a format the BitGo offline vault expects
+   * @param txInfo
+   * @param ethTx
+   * @param userKey
+   * @param backupKey
+   * @returns {{tx: *, userKey: *, backupKey: *, coin: string, amount: string, gasPrice: string, gasLimit: string, recipients: ({address, amount}|{address: ({address, amount}|string), amount: string}|string)[]}}
+   */
+  formatForOfflineVault(txInfo, ethTx, userKey, backupKey, gasPrice, gasLimit, callback) {
+    const backupHDNode = prova.HDNode.fromBase58(backupKey);
+    const backupSigningKey = backupHDNode.getKey().getPublicKeyBuffer();
+    return co(function *() {
```

Comment:
Is there a great reason you don't start the coroutine immediately, and instead do some assignments before? (These could be inside the coroutine)

Reply:
not a good reason, will change for clarity